Dockerfile-tests

```diff
@@ -0,0 +1,46 @@
+# We build an image with bee-queue's node_modules and all the source
+# files.  We jump through some Docker FROM hoops to avoid having the
+# bloat of the g++ compiler and other tools in the final image.
+
+FROM node:lts-alpine as beequeue-base
+
+# Info, and add bash, which we need in the final image
+RUN true \
+  && echo -n "node " && node --version \
+  && echo -n "npm " && npm --version \
+  && echo -n "apk " && apk --version \
+  && apk update \
+  && apk add --no-cache bash \
+  && rm -r /usr/share/man \
+  && true
+
+FROM beequeue-base as beequeue-node_modules
+
+# Ensure that tools for npm are available (for node-gyp in particular)
+# to build node_modules
+RUN true \
+  && apk add --no-cache python make g++ \
+  && true
+
+WORKDIR /bee-queue
+
+# Start with just these two files so as to optimize that npm install
+# only runs when package.json has been updated
+COPY package.json package-lock.json ./
+RUN npm --version && npm ci
+
+# Build the target image
+FROM beequeue-base
+
+WORKDIR /bee-queue
+
+COPY --from=beequeue-node_modules /bee-queue/node_modules ./node_modules/
+
+# We use a script for Docker-based testing (instead of npm test) so
+# that the dev cycle doesn't have to wait for the npm install
+# triggered by editing e.g. the scripts stanza in package.json
+ENV TEST_SCRIPT $TEST_SCRIPT
+CMD ./${TEST_SCRIPT}
```

Comment:
It seems to me like this is trivially overridden when running docker, but maybe it's a docker-compose limitation? I'd be in favor of hardcoding the test script for the `CMD`.

Reply:
Originally it was hard-wired, but that meant any new dev work required changing the one-true test script.  This way, the one-true script is the default, but dev can use any script.

Reply:
What I mean to say is that (as I understand it) when you use `docker run` you can just pass additional arguments to override the `CMD`.

Reply:
Yes, but it's a bit harder in the `docker-compose` setting.  And these are just initial versions -- I really don't have a good sense if anyone besides me is interested in this Dockerization effort.

I have a PR in the works that provides numerous cli arguments on the `go.sh` script which get communicated into the containers via env vars through `docker-compose`.

Reply:
:shrug: ok, thanks - haven't really used `docker-compose` much.